src/cargo/util/config.rs

```diff
@@ -190,16 +193,35 @@ impl Config {
             .map(AsRef::as_ref)
     }
 
+    /// Sets the clippy override. If this is true, clippy-driver is invoked instead of rustc.
+    pub fn set_clippy_override(&mut self, val: bool) {
+        self.clippy_override = val;
+    }
+
     /// Gets the path to the `rustc` executable.
     pub fn rustc(&self, ws: Option<&Workspace<'_>>) -> CargoResult<Rustc> {
         let cache_location = ws.map(|ws| {
             ws.target_dir()
                 .join(".rustc_info.json")
                 .into_path_unlocked()
         });
+        let wrapper = if self.clippy_override {
+            let tool = self.get_tool("clippy-driver")?;
+            let tool = paths::resolve_executable(&tool).map_err(|e| {
+                failure::format_err!("{}: please run `rustup component add clippy`", e)
+            })?;
+            if !paths::is_executable(&tool) {
```

Comment:
I don't know if this check is necesary, I feel like its unlikely that clippy-driver would exist and be resolved but would be a non executable file. The error message below is left silly as a filler because I'm not really sure what the response should be in this situation.